File: fs/BeginnersSelection/01_Product.fs
// https://atcoder.jp/contests/abs/tasks/abc086_a

// stdin.ReadLine()
// |> fun x -> x.Split(' ')
// |> Array.map (int >> (&&&) 1)
// |> Array.reduce (fun a b -> a &&& b)
// |> fun x -> (if x = 1 then "Odd" else "Even")
// |> printfn "%s"

// ----------

// stdin.ReadLine() |> fun x -> x.Split(' ') |> Array.map (int >> (&&&) 1) |> Array.reduce (fun a b -> a &&& b) |> fun x -> (if x = 1 then "Odd" else "Even") |> printfn "%s"

// https://atcoder.jp/contests/abs/submissions/9414632

// ----------

stdin.ReadLine()
|> fun x -> x.Split(' ')
|> Seq.map (int >> (&&&) 1)
|> Seq.reduce (&&&)
|> fun x -> (if x = 1 then "Odd" else "Even")
|> printfn "%s"

File: ex/BeginnersSelection/01_Product.ex
# https://atcoder.jp/contests/abs/tasks/abc086_a

use Bitwise

defmodule Main, do:
  def main(), do:
    IO.read(:line)
    |> String.trim
    |> String.split(" ")
    |> Enum.map(&String.to_integer/1)
    |> Enum.map(& &1 &&& 1)
    |> Enum.reduce(& &1 &&& &2)
    |> (& if &1 == 1, do: 'Odd', else: 'Even').()
    |> IO.puts

# https://atcoder.jp/contests/abs/submissions/21739209
